# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/other/RemoteAccessible.cpp
# Commit: 1a9901750064
# Full Hash: 1a9901750064de9fc00eaeb6094e639b035eb278
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2021-08-27 03:42:22
# Regressor Bug: 1727643
# File Overlap Count: 1
# Description:
#   Bug 1727643 - P3: Make numeric value methods abstract in Accessible and wire it thru. r=morgan
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123697
# ==============================================================================

diff -r c5789cd1f9c7 -r 1a9901750064 accessible/ipc/other/RemoteAccessible.cpp
--- a/accessible/ipc/other/RemoteAccessible.cpp	Thu Aug 26 19:40:05 2021 +0000
+++ b/accessible/ipc/other/RemoteAccessible.cpp	Thu Aug 26 19:40:05 2021 +0000
@@ -744,7 +744,11 @@
   Unused << mDoc->SendAtkKeyBinding(mID, &aBinding);
 }
 
-double RemoteAccessible::CurValue() {
+double RemoteAccessible::CurValue() const {
+  if (mCachedFields) {
+    return RemoteAccessibleBase<RemoteAccessible>::CurValue();
+  }
+
   double val = UnspecifiedNaN<double>();
   Unused << mDoc->SendCurValue(mID, &val);
   return val;
@@ -756,19 +760,31 @@
   return success;
 }
 
-double RemoteAccessible::MinValue() {
+double RemoteAccessible::MinValue() const {
+  if (mCachedFields) {
+    return RemoteAccessibleBase<RemoteAccessible>::MinValue();
+  }
+
   double val = UnspecifiedNaN<double>();
   Unused << mDoc->SendMinValue(mID, &val);
   return val;
 }
 
-double RemoteAccessible::MaxValue() {
+double RemoteAccessible::MaxValue() const {
+  if (mCachedFields) {
+    return RemoteAccessibleBase<RemoteAccessible>::MaxValue();
+  }
+
   double val = UnspecifiedNaN<double>();
   Unused << mDoc->SendMaxValue(mID, &val);
   return val;
 }
 
-double RemoteAccessible::Step() {
+double RemoteAccessible::Step() const {
+  if (mCachedFields) {
+    return RemoteAccessibleBase<RemoteAccessible>::Step();
+  }
+
   double step = UnspecifiedNaN<double>();
   Unused << mDoc->SendStep(mID, &step);
   return step;